### PR TITLE
Feat/integrate notifications controllers

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## Explanation

This adds some core controllers for the notification services that are consumed in both mobile and extension. (Maintained by the notifications team).

Controllers are:
- `ProfileSyncController` - which contains 2 controllers (`AuthenticationController` and `UserStorageController`)
  - These are used for adding or consuming authenticated endpoints and also make use of the profile syncing feature across platforms and devices.

- `NotificationServicesController` - which also contains 2 controllers (`NotificationServicesController` and `NotificationsServicesPushController`)
  - Both of these controllers are tightly coupled. The first manages pull based notifications and creation of resource; and the second is responsible for push notifications.

NOTE - these controllers should be treated as V0.x.x, as they are under development on extension and mobile (iron out any issues).

Previous PR: https://github.com/MetaMask/core/pull/4320
(closed this as it was getting to unruly to work in.)

## References

Here is a loom recording for the new controllers.
https://www.loom.com/share/4e95a8fcc2ae4d81b737265fc75571c5?sid=8247a832-3f2e-4837-ab74-30c45524ccfe

Future Improvements (as separate PRs)
1. Lets tidy up and improve Push Notifications for Mobile (after dog-fooding) - the implementation may differ on mobile.
2. I also want to improve imports so we can do paths (e.g. `@metamask/profile-sync-controller/sdk`) instead of the global named exports have.
3. Decoupling this from extension has shown some annoying messaging system actions back and forth (leading to some circular dependencies due to importing types). This has been resolved for now, but I want our team to rethink how we are using the messaging system to be more streamlined.
  - I don't want UserStorage controller calling actions to Notifications (we should not do this)
  - I don't really like how tied the notifications and push notifications do communication back and forth. I would rather we orchestrate communication in 1 controller (so communication is 1 way).

### `@metamask/profile-sync-controller`

- **ADDED**: new `AuthenticationController`
- **ADDED**: new `UserStorageController`

### `@metamask/notfication-services-controller`

- **ADDED**: new `NotificationServicesController`
- **ADDED**: new `NotificationServicesPushController`

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Review by matteoscurati — DISMISSED** [review]
Great job!

File: packages/profile-sync-controller/src/controllers/authentication/services.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        signature,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        raw_message: rawMessage,
+        metametrics: {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          metametrics_id: clientMetaMetricsId,
+          agent: 'extension',
```
Comment (Jonathansoufer) [inline packages/profile-sync-controller/src/controllers/authentication/services.ts:90]: shouldn't we receive the agent as param so we aren't tightly coupled to one single client/agent type?

Comment (Prithpal-Sooriya) [reply packages/profile-sync-controller/src/controllers/authentication/services.ts:90]: FFS, there are some other extension tie-ins as well that need replacing.
- Feature announcements - we don't want extension announcements
- Authentication - we want to say the agent is not just extension 

On it now.

**Review by Jonathansoufer — APPROVED** [review]
Great work!

**Review by Jonathansoufer — APPROVED** [review]
LGTM!

**Pull request merged**